### DISCUSSION TITLE
Micronaut consumes and produces stopped working after upgrade to micronaut 4

### DISCUSSION THREAD

**Discussion by jmmccota** [micronaut-projects/micronaut-core#10304] [General] [answered]
On micronaut version 3, it worked fine, while migrating to micronaut 4, stopped working

In a endpoint annotated with

     @Get("/{someId}/someAction", produces = ["application/some.v1+json"] )

When I call it (eg. postman) micronaut gives me an error on console

`io.micronaut.http.codec.CodecException: Cannot encode value [[]]. No possible encoders found for media type: application/vnd.v1+json`

here's a piece of my controller
    
```kotlin
@Validated
@Controller("/something")
class SomethingController(@Inject val service: SomeService) {

    @Secured(SecurityRule.IS_AUTHENTICATED)
    @Get("/{someId}/someAction", 
         produces = ["application/some.v1+json"], 
         consumes = ["application/some.v1+json"])
    fun getSomething(
            someId: UUID,
            principal: Principal
    ): List<SomeResource>  {
        principal as MyIdentity
        return service.findAll(someId, principal)
    }
```

My service has more endpoints, is there a way to write one codec that handle with all those customs produces and consumes? 

I posted first on [StackOverflow](https://stackoverflow.com/questions/77705544/micronaut-consumes-and-produces-stopped-working-after-upgrade-to-micronaut-4)

**Comment by yawkat** [comment, marked as answer]
try adding your type to additional-types as described here: https://docs.micronaut.io/latest/guide/index.html#consumesAnnotation

**Reply by jmmccota** [reply]
thanks, it worked!
